### PR TITLE
bug: fix HTTP response code type

### LINKED ISSUE #17: [BUG]: error_code is a string, not an integer
I guess this could be debatable if it is in fact a bug, but I think it is at least surprising.

In a `Lago::Api::HttpError`, the `#error_code` method returns a string, not an integer. This is apparently what `Net::Http` returns in its `response.code` method as it's the source of the value.

In every other ruby Http client that I've used (Faraday, Excon, etc) the code is an integer which allows matching against Rails' http symbols like `:not_found` etc.

Happy to provide a PR for this, either to change the type of the return value or at a minimum to document it in the README (or both)

### PR DESCRIPTION
**Pull request by prognostikos**
Ensure that Lago::Api::HttpError#error_code is an integer

Fixes #17

### CODE REVIEW

File: spec/lago/api/connection_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  context "when an unsuccessful request is made" do
+    before do
+      stub_request(:get, 'https://testapi.example.org/NOTFOUND')
+        .to_return(
+          body: JSON.generate({status: 404, error: "Not Found"}),
+          status: 404)
+    end
+
+    it "raises an exception with an integer error code" do
+      expect { connection.get('/NOTFOUND') }.to raise_error { |exception|
```
Comment (lovrocolic) [inline spec/lago/api/connection_spec.rb:22]: connection class does not have `get` method. We can maybe use `post` for testing purposes...

File: spec/lago/api/connection_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+RSpec.describe Lago::Api::Connection do
+  subject(:connection) do
+    described_class.new(
+      'fake-api-key',
+      URI('https://testapi.example.org/')
+    )
+  end
+
+  context "when an unsuccessful request is made" do
```
Comment (lovrocolic) [inline spec/lago/api/connection_spec.rb:13]: We prefer using single-quoted strings when there is no string interpolation. Can you please replace it in this file?

Comment (prognostikos) [reply spec/lago/api/connection_spec.rb:13]: 👍

File: spec/lago/api/connection_spec.rb (hunk 1 above)
Comment (prognostikos) [reply spec/lago/api/connection_spec.rb]: @lovrocolic I actually added `get` in #16 which is how I found this issue, guess I had already merged it in my fork.  But will update so the tests pass as is.

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**